### ISSUE TITLE
Bug: no-overwrite doesn't flag overwriting via destructuring assignment?

### ISSUE THREAD

**Issue by helixbass** [helixbass/eslint-plugin-coffee#31]
Looks like `no-overwrite` doesn't flag this:
```
itemVariable = null
grabsItem = do ->
  {id: itemVariable} = firstStatement.declarations[0]
```
